service/index/writer.go

```diff
@@ -259,12 +259,10 @@ func (w *Writer) apply(op func(*badger.Txn) error) error {
 func (w *Writer) done(err error) {
 
 	// When a transaction is fully committed, we get the result in this
-	// callback. If we have an error, we acquire the write lock for the error
-	// and store it.
+	// callback. In case of an error, we pipe it to the apply function through
+	// the error channel.
 	if err != nil {
-		w.Lock()
-		w.err = err
-		w.Unlock()
+		w.err <- err
```

Comment:
This does no longer replace any error value, so it can't interfere with logic in other places.